php-checks/src/main/java/org/sonar/php/checks/formatting/NamespaceAndUseStatementCheck.java

```diff
@@ -21,83 +21,108 @@
+import org.sonar.plugins.php.api.visitors.PHPVisitorCheck;
 
 import java.util.List;
 
-public class NamespaceAndUseStatementCheck {
+public class NamespaceAndUseStatementCheck extends PHPVisitorCheck implements FormatingCheck {
 
-  private List<AstNode> useNodes = Lists.newArrayList();
+  private static final String BLANK_LINE_NAMESPACE_MESSAGE = "Add a blank line after this \"namespace %s\" declaration.";
+  private static final String BLANK_LINE_USE_MESSAGE = "Add a blank line after this \"use\" declaration.";
+  private static final String USE_AFTER_NAMESPACE_MESSAGE = "Move the use declarations after the namespace declarations.";
+  private static final Kind[] USE_KINDS = {
+    Kind.USE_STATEMENT,
+    Kind.USE_CONST_STATEMENT,
+    Kind.USE_FUNCTION_STATEMENT};
 
-  public void visitNode(FormattingStandardCheck formattingCheck, AstNode node) {
-    if (node.is(PHPGrammar.NAMESPACE_STATEMENT)) {
-      checkBlankLineAfterNamespace(formattingCheck, node);
+  private List<UseStatementTree> useNodes = Lists.newArrayList();
+  private StatementTree nextStatement = null;
+  private FormattingStandardCheck check = null;
 
-    } else if (node.is(PHPGrammar.USE_STATEMENT)) {
-      useNodes.add(node);
-      AstNode nextNode = node.getNextAstNode().getFirstChild();
 
-      if (nextNode == null || nextNode.isNot(PHPGrammar.USE_STATEMENT)) {
-        checkUsesAreBeforeNamespace(formattingCheck, nextNode);
-        checkBlankLineAfterUses(formattingCheck, node);
-        useNodes.clear();
+  @Override
+  public void visitScript(FormattingStandardCheck formattingCheck, ScriptTree scriptTree) {
+    this.check = formattingCheck;
+
+    this.visitScript(scriptTree);
+
+    nextStatement = null;
+    useNodes.clear();
+  }
+
+  @Override
+  public void visitScript(ScriptTree tree) {
+    StatementTree previous = null;
+
+    for (StatementTree current : tree.statements()) {
+      nextStatement = current;
+
+      if (previous != null) {
+        previous.accept(this);
       }
+      previous = current;
     }
+    previous.accept(this);
   }
 
-  public void leaveFile() {
-    useNodes.clear();
+
+  @Override
+  public void visitNamespaceStatement(NamespaceStatementTree tree) {
+    if (check.hasNamespaceBlankLine && nextStatement != null && !isFollowedWithBlankLine(tree)) {
+      reportIssue(String.format(BLANK_LINE_NAMESPACE_MESSAGE, tree.namespaceName().fullName()), tree);
+    }
   }
 
-  private void checkBlankLineAfterUses(FormattingStandardCheck formattingCheck, AstNode useStatement) {
-    if (formattingCheck.hasUseBlankLine && isNotFollowedWithBlankLine(useStatement) && useStatement.getParent().getNextSibling() != null) {
-      formattingCheck.reportIssue("Add a blank line after this \"use\" declaration.", Iterables.getLast(useNodes));
+  @Override
+  public void visitUseStatement(UseStatementTree tree) {
+    useNodes.add(tree);
+
+    if (nextStatement != null && !nextStatement.is(USE_KINDS)) {
+        checkUsesAreBeforeNamespace();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this line start at column 7. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AIndentationCheck)